lib/metasploit/framework/command/console.rb

```diff
@@ -39,12 +39,50 @@ def spinner
     end
   end
 
+  def start_profiler
+    if ENV['METASPLOIT_CPU_PROFILE']
+      gem 'perftools.rb'
+      require 'perftools'
+
+      formatted_time = Time.now.strftime('%Y%m%d%H%M%S')
+      root = Pathname.new(__FILE__).parent
+      profile_pathname = root.join('tmp', 'profiles', 'msfconsole', formatted_time)
+
+      profile_pathname.parent.mkpath
+      PerfTools::CpuProfiler.start(profile_pathname.to_path)
+
+      at_exit {
+        PerfTools::CpuProfiler.stop
+
+        puts "Generating pdf"
+
+        pdf_path = "#{profile_pathname}.pdf"
+
+        if Bundler.clean_system("pprof.rb --pdf #{profile_pathname} > #{pdf_path}")
+          puts "PDF saved to #{pdf_path}"
+
+          Rex::Compat.open_file(pdf_path)
+        end
+      }
+    end
+
+    if ENV['METASPLOIT_MEMORY_PROFILE']
+      require 'memory_profiler'
+      MemoryProfiler.start
+      at_exit {
+        report = MemoryProfiler.stop
+        report.pretty_print(to_file: 'memory.profile')
+      }
+    end
+  end
+
   def start
     case parsed_options.options.subcommand
     when :version
       $stderr.puts "Framework Version: #{Metasploit::Framework::VERSION}"
     else
       spinner unless parsed_options.options.console.quiet
+      start_profiler
```

Comment:
Wondering if we should wrap this for `rescue LoadError` and notify unable to start profiler encase someone has env variables set in an install that does not supply the gems required?

Reply:
I don't understand what the purpose suggesting that we move the gems into developer tags was, if we now have to deal with the extra complexity of the gems probably not being there.

Surely it can't be space savings. This gem is 80K. The aws-* gems that are used by one module are almost 10MB.  Is there some other reason to add this complexity, or could I just move it to be a normal gem so anyone can memory profile in the field as well.

Reply:
Thinking about it, I'd advocate for making this something that can just be used by anyone at anytime, so move the gem to the normal group (we can still load on demand) and probably the CPU profiling ones as well. That would allow someone who is just noticing Metasploit doing something funny in either case the ability to supply debug info in case we cannot reproduce it locally. Sound good?

Reply:
I think that deserves its own discussion in another PR. I would suggest undoing the code move and contemplating what user-facing profiler would look like and how useful it would be later.

I don't think there needs to be `LoadError` rescues FWIW.

Reply:
All sounds reasonable, this was just suggestion based on scenarios possible.  Also note that anything added in `Gemfile` vs `.gemspec` adds different scoping requirements as well.

Reply:
Alright, I can move out the testing bits and just leave the other improvements